nncf/experimental/tensor/functions.py

```diff
@@ -0,0 +1,362 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+from typing import List, Optional, Tuple, TypeVar, Union
+
+from nncf.experimental.tensor import Tensor
+from nncf.experimental.tensor import unwrap_tensor_data
+from nncf.experimental.tensor.enums import TensorDataType
+from nncf.experimental.tensor.enums import TensorDeviceType
+
+TTensor = TypeVar("TTensor")
+
+
+def _tensor_guard(func: callable):
+    """
+    A decorator that ensures that the first argument to the decorated function is a Tensor.
+    """
+
```

Comment:
To save the name of the `func` function and the docstring
```suggestion
@functools.wraps(func)
```